spec/controllers/users_spec.rb

```diff
@@ -0,0 +1,62 @@
+ require_relative '../spec_helper'
+
+describe UsersController do
+  context 'with authenticated user logged in' do
```

Comment:
My rationale here is to prevent someone from typing in '/sign_up' into the address bar even if he already has an account and is signed in. In theory, if he is signed in, there should be no links on the pages to get to the sign up page -- but he could always type it in the address bar, right?